prime-router/src/main/kotlin/serializers/Hl7Serializer.kt

```diff
@@ -653,6 +661,31 @@ class Hl7Serializer(
         return message
     }
 
+    /**
+     * Loop through all [replaceValueMap] key value pairs to find the hl7Field
+     * if found, we check the value A is equal to the value need to replace.
+     * if so, we replace with value B.
+     * if all obove is not met, we return the same valueArg.
+     */
+    private fun replaceValueAwithB(
+        elementArg: Element,
+        replaceValueAwithBMap: Map<String, Any>,
+        valueArg: String
+    ): String {
+
+        replaceValueAwithBMap.forEach { element ->
+            @Suppress("UNCHECKED_CAST")
+            (element.value as ArrayList<Map<String, String>>).forEach { pairs ->
+                if (elementArg.hl7Field == element.key) {
+                    if (pairs.keys.first().trim() == valueArg)
+                        return pairs.values.first().trim()
+                }
```

Comment:
This doesn't check the hl7OutputFields. You have to check those in addition to the hl7Field

Reply:
Done.